src/config/wmodules-config.c

```diff
@@ -33,32 +44,56 @@ int Read_WModule(const OS_XML *xml, xml_node *node, void *d1, void *d2)
         return OS_INVALID;
     }
 
-    // Allocate memory
-
-    if ((cur_wmodule = *wmodules)) {
-        cur_wmodule_exists = *wmodules;
-        int found = 0;
+    cur_wmodule = *wmodules;
 
-        while (cur_wmodule_exists) {
-            if(cur_wmodule_exists->tag) {
-                if(strcmp(cur_wmodule_exists->tag,node->values[0]) == 0) {
-                    cur_wmodule = cur_wmodule_exists;
-                    found = 1;
-                    break;
-                }
+    // Wodles that are going to be enabled by default
+    int i=0;
+    while (default_modules[i] && !default_modules_initialized) {
+        if(!cur_wmodule) {
+            *wmodules = cur_wmodule = calloc(1, sizeof(wmodule));
+        } else {
+            os_calloc(1, sizeof(wmodule), cur_wmodule->next);
+            cur_wmodule = cur_wmodule->next;
+            if (!cur_wmodule) {
+                merror(MEM_ERROR, errno, strerror(errno));
+                return (OS_INVALID);
             }
-            cur_wmodule_exists = cur_wmodule_exists->next;
         }
+        // Point to read function
+        int (*function_ptr)(xml_node **nodes, wmodule *module) = default_modules[i];
+        
+        if(function_ptr(NULL, cur_wmodule) == OS_INVALID) {
+            return OS_INVALID;
+        }
+        i++;
+    }
 
-        if(!found) {
-            while (cur_wmodule->next)
-                cur_wmodule = cur_wmodule->next;
+    default_modules_initialized = true;
 
-            os_calloc(1, sizeof(wmodule), cur_wmodule->next);
-            cur_wmodule = cur_wmodule->next;
+
+    // Allocate memory
+    cur_wmodule_exists = *wmodules;
```

Comment:
Check if `cur_wmodule` is NULL and allocate the first element if so (https://github.com/wazuh/wazuh/pull/5760/files#diff-ce1bb33d396ffc9910cd9e8eef888ddbL61)